Source/WebCore/page/NavigationHistoryEntry.cpp

```diff
@@ -63,6 +68,30 @@ EventTargetInterface NavigationHistoryEntry::eventTargetInterface() const
     return NavigationHistoryEntryEventTargetInterfaceType;
 }
 
+uint64_t NavigationHistoryEntry::index() const
+{
+    RefPtr document = dynamicDowncast<Document>(scriptExecutionContext());
+    if (!document || !document->domWindow())
+        return -1;
```

Comment:
Return a std::optional<uint64_t> and return std::nullopt here?

Reply:
Unfortunately this is dictated by the spec.